src/main/java/space/morphanone/webizen/events/BasicRequestScriptEvent.java

```diff
@@ -181,12 +182,18 @@ else if (name.equals("query")) {
         else if (name.equals("query_map")) {
             MapTag mappedValues = new MapTag();
             String query = httpExchange.getRequestURI().getQuery();
-            if (query != null) {
-                for (String value : CoreUtilities.split(query, '&')) {
-                    List<String> split = CoreUtilities.split(value, '=');
-                    mappedValues.map.put(new StringHolder(split.get(0)), new ElementTag(split.get(1)));
+            try {
+                query = java.net.URLDecoder.decode(query, StandardCharsets.UTF_8.name());
```

Comment:
The decode has to be *after* splitting things, not before. Otherwise you ignore the entire point of encoding...